### PR TITLE
Update best-practices.md for HSTRING

### PR DESCRIPTION
**Pull request by AaronRobinsonMSFT**
Add recommendation for removed HSTRING support.

/cc @jkoritzinsky @elinor-fung

### CODE REVIEW

File: docs/standard/native-interop/best-practices.md [hunk 1]
```diff
@@ -211,6 +211,46 @@ A Windows `PVOID` which is a C `void*` can be marshaled as either `IntPtr` or `U
 
 [Data Type Ranges](/cpp/cpp/data-type-ranges)
 
+### Formally built-in supported types
```
Comment (gewarren) [inline docs/standard/native-interop/best-practices.md:214]: Did you mean "formerly"?

File: docs/standard/native-interop/best-practices.md [hunk 2]
```diff
@@ -211,6 +211,46 @@ A Windows `PVOID` which is a C `void*` can be marshaled as either `IntPtr` or `U
...
+
+There are rare instances when built-in support for a type is removed.
+
+The [`UnmanagedType.HString`](xref:System.Runtime.InteropServices.UnmanagedType) built-in marshal support was removed in the .NET 5 release. Binaries built that use this marshalling type and that target previous framework must be recompiled. Marshalling of this type is still possible but must be done manually. Below is an example that will work in moving forward and that is compatible with previous frameworks.
```
Comment (gewarren) [inline docs/standard/native-interop/best-practices.md:218]:
```suggestion
The [`UnmanagedType.HString`](xref:System.Runtime.InteropServices.UnmanagedType) built-in marshal support was removed in the .NET 5 release. You must recompile binaries that use this marshalling type and that target a previous framework. It's still possible to marshal this type, but you must marshal it manually, as the following code example shows. This code will work moving forward and is also compatible with previous frameworks.
```

**Review by gewarren — APPROVED** [review]
Thanks for adding this. I left some suggestions.

File: docs/standard/native-interop/best-practices.md (hunk 1 above)
Comment (AaronRobinsonMSFT) [reply docs/standard/native-interop/best-practices.md:214]: Yes. Thank you.

[the remark this answers is not in the document]
Comment (AaronRobinsonMSFT) [reply docs/standard/native-interop/best-practices.md]: Yep. @gewarren caught that. So much shame...

**Pull request merged**